sonar-php-plugin/src/main/java/org/sonar/plugins/php/PHPSensor.java

```diff
@@ -139,14 +135,11 @@ private static boolean inSonarQube(SensorContext context) {
     return !context.getSonarQubeVersion().isGreaterThanOrEqual(SQ_VERSION_6_0) || context.runtime().getProduct() != SonarProduct.SONARLINT;
   }
 
-  private void processCoverage(Map<String, Integer> numberOfLinesOfCode, SensorContext context) {
-    PhpUnitService phpUnitSensor = new PhpUnitService(
-      fileSystem,
-      new PhpUnitTestResultImporter(),
-      new PhpUnitCoverageResultImporter(fileSystem),
-      new PhpUnitItCoverageResultImporter(fileSystem),
-      new PhpUnitOverallCoverageResultImporter(fileSystem));
-    phpUnitSensor.execute(context, numberOfLinesOfCode);
+  private void processCoverage(SensorContext context, Map<String, Integer> numberOfLinesOfCode) {
+    final CompatibilityImportersFactory importersFactory = new CompatibilityImportersFactory(context);
+    importersFactory.deprecationWarnings().forEach(LOG::warn);
```

Comment:
I would rather keep the importers factory as context-neutral and as technology-neutral as possible and the act of logging is a technicality that is specific to this usage context. In other words, `CompatibilityImportersFactory` is a pure domain object right now with only business logic and very easy to test, I would like to keep it like that.